torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -295,7 +272,21 @@ def _call_function_with_auto_output_flattening(
         ),
         example_value=flat_example_value,
     )
+    return flat_variable
+
+
+def overwrite_tensor_vt_requires_grad(graph_output_vts, flat_variable):
+    # this is required for faithfully representing the autograd.Function forward
+    # outputs.
```

Comment:
nit: a bit more explanation, "all outputs of autograd.Function have requires_grad=True. We turn off grad_mode in autograd.Function, so our outputs naively have requires_grad=False. So we hackily force them back on here. A better solution would be X but we decided it wasn't worth it and likely won't run into problems"